### DISCUSSION TITLE
Issues when trying to configure Huawei device

### DISCUSSION THREAD

**Discussion by Vaalen** [scrapli/scrapligo#198] [Q&A] [answered]
Getting the below error when trying to use "*.SendConfig()" towards a Huawei NE40 router.

Using this platform:
assets/platforms/huawei_vrp.yaml

Error:
Failed run command; error: errPrivilegeError: requested target privilege level 'configuration' is not a valid privilege level

Guessing the code is trying to get to "configuration" level, but it's not called that in Huawei devices.

**Comment by carlmontanari** [comment]
the [platform](https://github.com/scrapli/scrapligo/blob/main/assets/platforms/huawei_vrp.yaml) has no configuration privilege level, so dont use *Config methods.

**Reply by Vaalen** [reply]
So do you recommend to just use *.SendCommand() instead?
Because when I use that I get timeouts instead.

I can post my short sample code for this if you want?

**Reply by Vaalen** [reply]
I'm no expert coder, but in the NE40 platform the "system-view" level is comparable to the "configuration" level of Cisco IOS XR devices. So I'm guessing the issue would be resolved by matching "system-view" to be a configuration level somehow?

**Reply by carlmontanari** [reply, marked as answer]
then use sendconfig with the appropriate [privilege level](https://github.com/scrapli/scrapligo/blob/main/driver/opoptions/network.go#L11)

**Reply by Vaalen** [reply]
Tried it out, it doesn't work with sendconfig + opoptions, but it works with sendcommand + opoptions.
